Implement support for big-endian hosts

zeldin opened the pull request:
This can also serve as a base for simulating big- or bi-endian RISC-V on little-endian hosts.

Tested on POWER9 using integer, floating point and vector operations.

aswaterman commented:
Given that Spike doesn't actually support big-endian targets yet, the ELF loader should not be changed to accept big-endian ELFs.  Failing to issue an error for big-endian ELFs will lead to confusion.

zeldin commented:
@aswaterman A fair point.  (Although BE ELFs weren't exactly rejected before, they just gave the
loader a high chance of bombing out due to misinterpreting the fields. :smile_cat:)

I changed the assert to reject BE ELFs and removed the BE cases at the bottom.  It's easy enough to put them back when needed.

aswaterman commented:
Ah, you’re right. Well, thanks for improving the status quo :-)

On Mon, Aug 19, 2019 at 10:10 AM Marcus Comstedt <notifications@github.com>
wrote:

> @aswaterman <https://github.com/aswaterman> A fair point. (Although BE
> ELFs weren't exactly rejected before, they just gave the
> loader a high chance of bombing out due to misinterpreting the fields. 😸)
>
> I changed the assert to reject BE ELFs and removed the BE cases at the
> bottom. It's easy enough to put them back when needed.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/riscv/riscv-isa-sim/pull/320?email_source=notifications&email_token=AAH3XQUDQ3IDY6225Z43THLQFLHXJA5CNFSM4IMTGPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4TUY3I#issuecomment-522669165>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAH3XQRJRHEXPY7Y4KGUHQDQFLHXJANCNFSM4IMTGPAQ>
> .
>

zeldin commented:
Rebased and fixed the conflict in `mmu.h`
